src/Illuminate/Support/Arr.php

```diff
@@ -116,6 +116,36 @@ public static function dot($array, $prepend = '')
     }
 
     /**
+     * Expand a dotted array. Acts the opposite way of Arr::dot().
+     *
+     * @param  array  $array
+     * @param  bool  $recursively
+     * @return array
+     */
+    public static function unDot($array, $recursively = true)
```

Comment:
Instead of `$recursively`, wouldn't it make sense to accept a `$depth`? If can default to `INF`.